File: 3217-delete-nodes-from-linked-list-present-in-array/3217-delete-nodes-from-linked-list-present-in-array.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        unordered_set<int> s;
        int n=nums.size();
        for(int i=0;i<n;i++){
            s.insert(nums[i]);
        }
        ListNode * ans=head;        
        while(ans!=NULL && s.find(ans->val)!=s.end()){
            ans=ans->next;       //Check for the head node excusively
        }
        head=ans;
        while(head!=NULL && head->next!=NULL){
            if(s.find(head->next->val)!=s.end()){
                head->next=head->next->next;
                continue;
            }
            head=head->next;
        }
        return ans;
    }
};

File: 3217-delete-nodes-from-linked-list-present-in-array/3217-delete-nodes-from-linked-list-present-in-array.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {number[]} nums
 * @param {ListNode} head
 * @return {ListNode}
 */
var modifiedList = function(nums, head) {
    let node = new ListNode(0, head), numsSet = new Set(nums)
    while(node.next !== null) {
        let val = node.next.val
        if (numsSet.has(val)) {
            if (node.next === head) head = head.next
            node.next = node.next.next
        } else node = node.next
    }
    return head
};